clang/docs/analyzer/checkers.rst

```diff
@@ -3138,10 +3138,13 @@ are detected:
   allowed in this state.
 * Invalid 3rd ("``whence``") argument to ``fseek``.
 
-The checker does not track the correspondence between integer file descriptors
-and ``FILE *`` pointers. Operations on standard streams like ``stdin`` are not
-treated specially and are therefore often not recognized (because these streams
-are usually not opened explicitly by the program, and are global variables).
+The checker assumes by default that any of the stream operations, except the
+write operations (like ``fwrite``, ``fprintf``) can fail. The option
+``Pedantic`` controls if the write operations are assumed to be successful or
+not. If set to ``true``, the write operations (including change of file
+position) can fail too (default is ``false``). This option was added to prevent
+many checker warnings if the code contains many write operations without error
+check.
```

Comment:
```suggestion
When this checker models a stream operation, it usually splits the analysis into
two cases: one where the stream operation succeeds, and one where it fails.
However, in the case of write operations (like ``fwrite``, ``fprintf`
and even `fsetpos`) this behavior could produce a large amount of unwanted
reports on projects that don't have error checks around the write operations --
so by default the checker assumes that write operations always succeed.

This behavior can be controlled by the ``Pedantic`` flag: with
``-analyzer-config alpha.unix.Stream:Pedantic=true`` the checker will model the
cases where a write operation fails and report situations where this leads to
erroneous behavior. (The default is ``Pedantic=false``, where write operations
are assumed to succeed.)
```
I felt that this paragraph was a bit difficult to understand, so I rewrote it in a somewhat more verbose and hopefully clearer way. Feel free to tweak this if you feel that there could be further improvements.